pkg/apis/serving/v1/revision_lifecycle.go

```diff
@@ -51,6 +51,8 @@ const (
 	ReasonProgressDeadlineExceeded = "ProgressDeadlineExceeded"
 )
 
+// RevisionConditionActive is not part of the RevisionConditionSet because we can have Inactive Ready Revisions (scale to zero)
+// TODO: add to the revisionCondSet or at least initialize RevisionConditionActive and deal with the edge cases
```

Comment:
Is this TODO meant to be tackled by this PR?

Reply:
No, but let me create an issue to discuss it. Right now is working fine as it is so is worth to leave it for further discussion.

Reply:
Removed the last part since it was confusing. But yes, I meant that if the `RevisionConditionActive` was `nil` by the time it got to this check: https://github.com/knative/serving/blob/4c39ae2a139b06baadefc0ce4078223899522f88/pkg/reconciler/revision/reconcile_resources.go#L77 it was `Pod Forbidden..`, and if it was not `nil` (the config sometimes initialize it) then it'll wait til `ProgressDeadline` and never surface the error to the `Service Status`

Reply:
Let's capture this in an issue and not as a comment - those usually get lost in time.

Reply:
I believe #14513 was created to address this

Reply:
Yep closed since it has been answered by @dprotaso ! Thanks